sycl/source/detail/global_handler.cpp

```diff
@@ -27,6 +27,39 @@
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
+
+// Utility class to track references on object.
+// Used for Scheduler now and created as thread_local object.
+// Origin idea is to track usage of Scheduler from main and other used threads -
+// they increment MCounter; and to use but not add extra reference by our
+// thread_pool threads. For this control MIncrementCounter class member is used.
+// MObj and MReleaseCalled is extra protection needed to handle case when main
+// thread finished but thread_pool is still running and we will join that
+// threads in releaseResources call.
```

Comment:
```suggestion
```

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/71e9048af88ddcc853968cf9bbaa050f63ad323e